src/conversion.c

```diff
@@ -405,11 +405,12 @@ int igraph_to_directed(igraph_t *graph,
         IGRAPH_CHECK(igraph_create(&newgraph, &edges,
                                    (igraph_integer_t) no_of_nodes,
                                    IGRAPH_DIRECTED));
-        IGRAPH_FINALLY(igraph_destroy, &newgraph);
-        igraph_vector_destroy(&edges);
+        IGRAPH_FINALLY(igraph_destroy, &newgraph);        
         IGRAPH_I_ATTRIBUTE_DESTROY(&newgraph);
         IGRAPH_I_ATTRIBUTE_COPY(&newgraph, graph, 1, 1, 1);
+        igraph_vector_destroy(&edges);
         IGRAPH_FINALLY_CLEAN(2);
+
         igraph_destroy(graph);
```

Comment:
Should this `igraph_destroy(graph)` not be located before `IGRAPH_FINALLY_CLEAN`? Well, actually, it shouldn't matter much, we can pop from the stack before destroying it, but we almost always first destroy and then pop from the stack.

Reply:
Technically it doesn't matter, as `igraph_destroy` doesn't fail.

Semantically, I think it's correct to keep `igraph_destroy(igraph)` at the end. The finally stack contains not `igraph_destroy(graph)`, but `igraph_destroy(newgraph)`. This is what I call the "constructor-finally" (when I think about it). It's `FINALLY_CLEAN` has no corresponding explicit `destroy` call. In constructors, the idea is that we put a `destroy` call on the finally stack in case the constructor fails. But if the constructor succeeds, of course its result should not be destroyed.

Reply:
I think it's more readable to keep it at the end. If I put it the `igraph_destroy(graph)` right before the `FINALLY_CLEAN`, it makes the impression to the reader that it is an `igraph_destroy(graph)` that is being cleaned from the stack—this is not correct. 

The mismatch between the number of destroy calls precending the `FINALLY_CLEAN` and the number in the `FINALLY_CLEAN` is expected, for the same reason as in constructors.

Reply:
Ah yes, you are right, you only destroy the old allocated `graph`, which you then supply with the new allocated stuff from `newgraph`.